### PR TITLE
Fix 8064: Some v2 updates in NFT details

### LINKED ISSUE #8064: Wallet: NFT Details V2
```[tasklist]
### Tasks
- [x] New layout + network and account icons
- [x] Send/Hide/Mark as not junk/Remove
- [x] Remove confirmation pop-up
- [x] Updates on Navigation Bar
```

### PR DESCRIPTION
**Pull request by nuo-xu**
## Summary of Changes

This pull request fixes #8064

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`
- [x] New or updated UI has been tested across:
  - [x] Light & dark mode
  - [x] Different size classes (iPhone, landscape, iPad)
  - [x] Different dynamic type sizes

## Test Plan:
1. Set up wallet and make sure add or auto-discover some NFTs
2. Click any NFT to go to its details screen 
3. The NFT image now has a network icon and an account blockie that owns this NFT (This is missing when you come from asset search list. Not listed in the figma, if this is required we can add later)
4. On top right there is a 3-dot button which will give you the same options when you long-press the NFT grid in the previous screen + a new option `Send` if this NFT is visible 
5. Click send will bring you to the send screen
6. Click hide will update NFT visible status which will change the options of the 3-dot, and move the NFT to the `Hide` selection if you go back to the previous screen
7. Click unhide will update NFT visible status which will change the options of the 3-dot, and move the NFT to the `Collected` selection if you go back to the previous screen.
8. Click mark as not junk will update NFT spam status which will change the options of the 3-dot, and move the NFT to the `Collected` selection if you go back to the previous screen
9. Click Don't show in wallet will remove NFT from the wallet after you confirm to remove it. You will be pop-off from the details screen and NFT will be remove from the wallet entirely. 
10. Below the image, we have `Overview`, `Description` and `Properties` sections(optional), if we have information from NFT's metadata. 

## Screenshots:

Simulator Screenshot - iPhone 15 Pro - 2023-11-14 at 17 47 09 | Simulator Screenshot - iPhone 15 Pro - 2023-11-14 at 17 47 44
--- | ---
Simulator Screenshot - iPhone 15 Pro - 2023-11-14 at 17 47 13 | Simulator Screenshot - iPhone 15 Pro - 2023-11-14 at 17 47 47
Simulator Screenshot - iPhone 15 Pro - 2023-11-14 at 17 47 19 | Simulator Screenshot - iPhone 15 Pro - 2023-11-14 at 17 47 41
Simulator Screenshot - iPhone 15 Pro - 2023-11-14 at 17 48 49 | Simulator Screenshot - iPhone 15 Pro - 2023-11-14 at 21 09 57

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: Sources/BraveWallet/Crypto/NFT/NFTDetailView.swift [hunk 1]
```diff
@@ -195,6 +225,105 @@ struct NFTDetailView: View {
...
+          }
+        } label: {
+          Label(Strings.Wallet.otherWalletActionsAccessibilityTitle, braveSystemImage: "leo.more.horizontal")
+            .labelStyle(.iconOnly)
+            .foregroundColor(Color(.braveBlurpleTint))
+        }
+      }
+    }
+    .background(
+      WalletPromptView(
```
Comment (StephenHeaps) [inline Sources/BraveWallet/Crypto/NFT/NFTDetailView.swift:262]: `isPresentingRemoveAlert` needs reset to false when wallet is auto-locked, we have this issue on NFTs grid view also: https://github.com/brave/brave-ios/issues/8412.

PS this PR will need rebased with latest development & use `isWalletLocked` (This branch out of date currently as it's still using `KeyringInfo`) 🙂.

File: Sources/BraveWallet/Crypto/NFT/NFTDetailView.swift [hunk 2]
```diff
@@ -71,121 +84,138 @@ struct NFTDetailView: View {
...
+                  if let explorerURL = nftDetailStore.networkInfo.blockExplorerUrls.first, let nftURL = URL(string: "\(explorerURL)/address/\(nftDetailStore.nft.contractAddress)") {
+                    openWalletURL(nftURL)
                   }
-                  .foregroundColor(Color(.braveBlurple))
                 }
               }
+            } label: {
+              Text(nftDetailStore.nft.contractAddress.truncatedAddress)
+                .font(.subheadline)
+                .foregroundColor(Color(.braveBlurpleTint))
```
Comment (StephenHeaps) [inline Sources/BraveWallet/Crypto/NFT/NFTDetailView.swift]: I think we'd want to use `Label` with `arrow.up.forward.square` image here, similar to linking out to transactions? I _believe_ the only spot we don't include this icon is learn more links in settings.

File: Sources/BraveWallet/Crypto/NFT/NFTDetailView.swift [hunk 3]
```diff
@@ -71,121 +84,138 @@ struct NFTDetailView: View {
...
-                      Image(systemName: "arrow.up.forward.square")
-                    }
-                    .foregroundColor(Color(.braveBlurple))
-                  } else {
-                    HStack {
-                      Text("\(nftDetailStore.nft.name) #\(nftDetailStore.nft.tokenId)")
-                      Image(systemName: "arrow.up.forward.square")
-                    }
-                    .foregroundColor(Color(.braveBlurple))
+                    nftURL = URL(string: "\(baseURL)?a=\(tokenId)")
```
Comment (StephenHeaps) [inline Sources/BraveWallet/Crypto/NFT/NFTDetailView.swift]: I think this might be beneficial in an extension on `NetworkInfo` (including the non-test network block below)? Similar to `func txBlockExplorerLink(txHash:for:) -> URL?`

File: Sources/BraveWallet/Crypto/Stores/NFTDetailStore.swift [hunk 4]
```diff
@@ -48,26 +54,43 @@ struct NFTMetadata: Codable, Equatable {
...
+    case value
+  }
+}
+
 class NFTDetailStore: ObservableObject, WalletObserverStore {
+  private let assetManager: WalletUserAssetManagerType
   private let rpcService: BraveWalletJsonRpcService
   private let ipfsApi: IpfsAPI
-  let nft: BraveWallet.BlockchainToken
+  let owner: BraveWallet.AccountInfo?
```
Comment (StephenHeaps) [inline Sources/BraveWallet/Crypto/Stores/NFTDetailStore.swift]: What about the case where `NFTStore` hasn't fetched balance of NFT's? I don't think we need to fetch balance in `NFTStore` when `Hide Unowned` is disabled and we are not grouping by `Accounts`, but looks like we [regressed with NFT Grouping](https://github.com/brave/brave-ios/blob/099af6ea2f5161b7fa0c0ebb12be3385c052252a/Sources/BraveWallet/Crypto/Stores/NFTStore.swift#L138-L139) there? I think we likely want to optimize that again in a separate PR (faster loading of NFT grid), but this would also apply to the case of slower internet and opening NFT Detail prior to balance fetch returning.

We might need to fetch balance in this store, but that should also solve NFT Detail opened from asset search not showing owner blockie (if that is needed).

File: Sources/BraveWallet/Extensions/BraveWalletExtensions.swift [hunk 5]
```diff
@@ -309,7 +309,7 @@ extension BraveWallet.BlockchainToken {
   
   var nftTokenTitle: String {
     if isErc721, let tokenId = Int(tokenId.removingHexPrefix, radix: 16) {
-      return "\(name) #\(tokenId)"
+      return "\(symbol) #\(tokenId)"
```
Comment (StephenHeaps) [inline Sources/BraveWallet/Extensions/BraveWalletExtensions.swift]: Is this change intended? This changes more than just NFT Detail view.

File: Sources/BraveWallet/WalletStrings.swift [hunk 6]
```diff
@@ -3754,6 +3747,27 @@ extension Strings {
...
+      tableName: "BraveWallet",
+      bundle: .module,
+      value: "Properties",
+      comment: "The section header that displays the all the properties/attributes information of this NFT."
+    )
+    public static let nftDetailOwnedBy = NSLocalizedString(
+      "wallet.nftDetailOwnedBy",
+      tableName: "BraveWallet",
+      bundle: .module,
+      value: "Owned by",
```
Comment (StephenHeaps) [inline Sources/BraveWallet/WalletStrings.swift]: `Token Standard` updated to title-case, but this is sentence case (and `Contract Address` is title-case). I'm not sure if title-case or sentence case is intended here?

File: Sources/BraveWallet/Crypto/NFT/NFTDetailView.swift (hunk 2 above)
Comment (nuo-xu) [reply Sources/BraveWallet/Crypto/NFT/NFTDetailView.swift]: hmmm dont have that image in the design. but i see desktop has it. will add one for iOS

File: Sources/BraveWallet/Extensions/BraveWalletExtensions.swift (hunk 5 above)
Comment (nuo-xu) [reply Sources/BraveWallet/Extensions/BraveWalletExtensions.swift]: on Figma, it's supposed to be `symbol+token id`
before erc721 `metadata.name` does not have any token id information. but now it does. So lot's of place is display `xxxxx #123 #123` with double token id. (we can fix it later if necessary)
I will keep this as the old one and create a new one just for NFT details.

File: Sources/BraveWallet/WalletStrings.swift (hunk 6 above)
Comment (nuo-xu) [reply Sources/BraveWallet/WalletStrings.swift]: ~~i thought `Owned by` title-case is `Owned by`?~~ Edit: Nvm, i found `Sort Notes By`
checked Apple has
1. StandBy 
2. Show as Overdue
3. Use Volume Up for Burst
4. Organize by Thread

File: Sources/BraveWallet/Crypto/Stores/NFTDetailStore.swift (hunk 4 above)
Comment (nuo-xu) [reply Sources/BraveWallet/Crypto/Stores/NFTDetailStore.swift]: updated [883a39f](https://github.com/brave/brave-ios/pull/8408/commits/883a39feda833d08a48a172d39d10dc4d0f63cb7) 
only fetching balance when `Hide Unowned` is on in `NFTStore`.
fetching NFT balance in NFT Details if `owner` is nil -> this will bring the owner account info when user comes from asset search.

File: Sources/BraveWallet/Crypto/Stores/NFTStore.swift [hunk 7]
```diff
@@ -303,42 +303,44 @@ public class NFTStore: ObservableObject, WalletObserverStore {
...
-                  for account in allAccounts where account.coin == nft.coin {
-                    group.addTask { @MainActor in
-                      let balanceForToken = await rpcService.balance(
-                        for: nft,
-                        in: account,
-                        network: networkForNFT
-                      )
-                      return [account.address: Int(balanceForToken ?? 0)]
+      // if we're not hiding unowned or grouping by account, balance isn't needed
+      if filters.isHidingUnownedNFTs {
```
Comment (StephenHeaps) [inline Sources/BraveWallet/Crypto/Stores/NFTStore.swift]: nit: I think we also want to include `|| filters.groupBy == .accounts`, in case we later remove logic that sets `isHidingUnownedNFTs` to true when grouping my accounts? Presumably would get caught in that PR though.

File: Sources/BraveWallet/Extensions/BraveWalletExtensions.swift [hunk 8]
```diff
@@ -259,6 +261,38 @@ extension BraveWallet.NetworkInfo {
     }
     return nil
   }
+  
+  /// Generate the explorer link for the given ERC721 token with the current NetworkInfo
+  func erc721TokenBlockExplorerURL(_ token: BraveWallet.BlockchainToken) -> URL? {
```
Comment (StephenHeaps) [inline Sources/BraveWallet/Extensions/BraveWalletExtensions.swift]: nit: I think we can combine the two functions? Both pass the same info to the same object 🙂.
`erc721TokenBlockExplorerURL` & `splTokenBlockExplorerURL`

File: Sources/BraveWallet/Crypto/NFT/NFTDetailView.swift [hunk 9]
```diff
@@ -60,129 +78,237 @@ struct NFTDetailView: View {
...
-        if isSVGImage {
-          Text(Strings.Wallet.nftDetailSVGImageDisclaimer)
-            .multilineTextAlignment(.center)
-            .font(.footnote)
-            .foregroundColor(Color(.secondaryBraveLabel))
-            .frame(maxWidth: .infinity)
+        .listRowBackground(Color(.secondaryBraveGroupedBackground))
+      } header: {
+        Text(Strings.Wallet.nftDetailOverview)
+          .listRowInsets(.zero)
```
Comment (StephenHeaps) [inline Sources/BraveWallet/Crypto/NFT/NFTDetailView.swift]: I don't think we want `listRowInsets(.zero)` on any of these headers.
NFT Detail | Settings / Default
--|--
NFT Detail | Settings : default

File: Sources/BraveWallet/Crypto/NFT/NFTDetailView.swift [hunk 10]
```diff
@@ -60,129 +78,237 @@ struct NFTDetailView: View {
...
+      }
+      if let nftMetadata = nftDetailStore.nftMetadata, let description = nftMetadata.description, !description.isEmpty {
+        Section {
+          Text(description)
+            .font(.subheadline)
+            .foregroundColor(Color(.braveLabel))
+            .listRowBackground(Color(.secondaryBraveGroupedBackground))
```
Comment (StephenHeaps) [inline Sources/BraveWallet/Crypto/NFT/NFTDetailView.swift:170]: Seems to be something off with the padding on iPad here, but iPhone seems okay 🤔
iPad | iPhone
--|--

Comment (nuo-xu) [reply Sources/BraveWallet/Crypto/NFT/NFTDetailView.swift:170]: seems like its only happens on iOS15 both iphone and ipad.

Comment (StephenHeaps) [reply Sources/BraveWallet/Crypto/NFT/NFTDetailView.swift:170]: It's not only on iOS 15
Screenshot 2023-11-16 at 10 04 12 AM

Comment (nuo-xu) [reply Sources/BraveWallet/Crypto/NFT/NFTDetailView.swift:170]: i was not able to reproduce every time. but when i was, it's always iOS15 only. 
so i addd the row insect for iPad only since you are able to reproduce on iOS17. Please try again, hope it fixes

**Review by StephenHeaps — APPROVED** [review]
